README.md

```diff
@@ -0,0 +1,113 @@
+# Action Server
+The action server is an actionlib server for managing the execution of high
+level tasks and their semantic chaining. It takes actionlib goals describing
+series of high level tasks in json form, checks the consistency of the
+semantics, and executes them. This readme explains the architecture of the
+action server, the procedure of handling a goal and the lifecycle of an action.
+
+## Architecture
+
+![Action Server architecture](doc/action_server_architecture.jpg)
+
+The action server uses an actionlib interface for communication with its
+clients. A client implementation is provided as abstraction from the actionlib
+client. Current examples of clients to the action server are:
+ - The GPSR challenge
+ - The Natural Language Console
+Consult one of these implementations for example usage of the Client class.
```

Comment:
Probably, these are relatively complex implementations. A very very basic example client would be very nice, e.g. to inspect the cabinet or something.

Reply:
I think that would clutter the package too much, but I'll add an example in the docstring of the Client class.